core/src/main/java/io/questdb/griffin/ExpressionParser.java

```diff
@@ -410,7 +418,48 @@ void parseExpr(GenericLexer lexer, ExpressionParserListener listener, SqlParserC
                         }
 
                         break;
-
+                    case 'd':
+                    case 'D':
+                        if (prevBranch == BRANCH_LEFT_PARENTHESIS && SqlKeywords.isDistinctKeyword(tok)) {
+                            // rewrite count(distinct x) to count_distinct(x)
+                            // and string_agg(distinct x) to string_distinct_agg(x)
+                            if (opStack.size() > 1) {
+                                ExpressionNode en = opStack.peek();
+                                // we are in the BRANCH_LEFT_PARENTHESIS, so the previous token must be
+                                // a control token '('
+                                assert Chars.equals(en.token, '(') && en.type == ExpressionNode.CONTROL;
+                                CharSequence tokenStash = GenericLexer.immutableOf(tok);
+                                CharSequence nextToken = SqlUtil.fetchNext(lexer);
+                                if (nextToken != null) {
+                                    // if distinct is the only argument then we treat it as a column name
+                                    // and not a keyword. strictly speaking, keywords used as column names
+                                    // should be quoted, but for now we are not enforcing it here for backwards compatibility reasons.
+                                    // it's something we could/should consider in the future
+                                    if (!Chars.equals(nextToken, ')')) {
+                                        en = opStack.peek(1);
+                                        if (en.type == ExpressionNode.LITERAL) {
+                                            if (Chars.equalsIgnoreCase("count", en.token)) {
```

Comment:
nit:
```suggestion
                                            if (SqlKeywords.isCountKeyword(en.token)) {
```

Reply:
this was my original code. 
but then I figured out it would be misleading - since in this context it's not a keyword. it's a function name. wdyt?

Reply:
I don't mind keeping it as is. BTW are there any SQL clauses that use `count` as a keyword? My impression is that it's only used as a function name.

Reply:
> BTW are there any SQL clauses that use count as a keyword? 
that's an excellent question!